src/wallet/wallet.cpp

```diff
@@ -1406,9 +1406,19 @@ bool CWallet::ImportScriptPubKeys(const std::string& label, const std::set<CScri
         return false;
     }
     AssertLockHeld(spk_man->cs_wallet);
-    if (!spk_man->ImportScriptPubKeys(label, script_pub_keys, have_solving_data, apply_label, timestamp)) {
+    if (!spk_man->ImportScriptPubKeys(script_pub_keys, have_solving_data, timestamp)) {
         return false;
     }
+    if (apply_label) {
+        WalletBatch batch(*database);
```

Comment:
26fef3a525fd98f8205fd2f859f476fbf2cc6fcd

Just noting that there's actually a behavior change here - now if `LegacyScriptPubKeyMan::ImportScriptPubKeys` fails __no label is applied__.

Also, another change, which I think is harmless, is that now 2 batches are always used - easily "reverted" by passing the batch to `LegacyScriptPubKeyMan::ImportScriptPubKeys` from this method.

Reply:
Those are harmless. `ImportScriptPubKeys` only fails when writing to the database fails, so you'll have bigger problems if that happens.